rmf_traffic_editor/gui/delete_dialog.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2019 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "delete_dialog.h"
+#include <algorithm>
+#include <QtWidgets>
+using std::vector;
+using std::string;
+
+
+DeleteDialog::DeleteDialog(
+  QWidget* parent,
+  Building& building,
+  int level_idx,
+  int selected_vertex_idx)
+: QDialog(parent)
+{
+
+
+  setWindowTitle("Warning");
+  QVBoxLayout* warning_message_vbox_layout = new QVBoxLayout;
+
+  // check if vertex is a lift cabin
+  std::vector<std::string> lift_levels;
+  auto level = building.levels[level_idx];
+  const auto v = level.vertices[selected_vertex_idx];
+  auto it = v.params.find("lift_cabin");
+  if ((it != v.params.end()))
+  {
+    auto lift = building.get_lift(v.lift_cabin());
+
+    for (const auto& level_door : lift.level_doors)
+    {
+      lift_levels.push_back(level_door.first);
+    }
+
+    std::string s = lift_levels[0];
```

Comment:
Spooky, this segfaults if `lift_levels` is empty, which can happen if a user disables all floors and then tries to delete a vertex

Reply:
I've changed the string concatenating logic. It should be safe now.